R/brace_linter.R

```diff
@@ -12,6 +12,36 @@
 #'
 #' @param allow_single_line if `TRUE`, allow an open and closed curly pair on the same line.
 #'
+#' @examples
+#' library(lintr)
+#'
+#' # will produce lints
+#' lint(
+#'   text = "f <- function() { 1 }",
+#'   linters = brace_linter()
+#' )
+#'
+#' lint(
+#'   text = "if (TRUE) {
```

Comment:
feels _slightly_ misleading since the indentation of the actual code as parsed won't match what you see visually, but I don't have a good answer for how to fix it... WDYT

Reply:
Yeah, that's a good point.

How about we first show them what the code looks like and then use that text in the `lint()` call?

``` r
library(lintr)

cat("if (TRUE) { \nreturn(1) }")
#> if (TRUE) { 
#> return(1) }

lint(
  text = "if (TRUE) { \nreturn(1) }",
  linters = brace_linter()
)
#> <text>:2:11: style: [brace_linter] Closing curly-braces should always be on their own line, unless they are followed by an else.
#> return(1) }
#>           ^
```

<sup>Created on 2022-10-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
good idea, LGTM

Reply:
Done.